oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java

```diff
@@ -354,6 +353,8 @@ public APIGatewayProxyResponseEvent handleRequest(
 
                 Boolean newAccount = userInfo.getBooleanClaim("new_account");
                 AccountState accountState = newAccount ? AccountState.NEW : AccountState.EXISTING;
+
+                sessionService.storeOrUpdateSession(userSession.setNewAccount(accountState));
```

Comment:
Does `userSession.setNewAccount(accountState)` return back `accountState` . I would have though that we would want `sessionService.storeOrUpdateSession(accountState)`

Reply:
SetNewAccount modifies the session object and returns it on one go, the `sessionService.storeOrUpdateSession` takes a session object as it's argument. The service is separate from the individual session so we need to pass the updated session object here and can't just pass the account state